### PR TITLE
feat: add ethereum to leverage ui

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds Ethereum support to Leverage UI
* Fixes arrow alignment in tx review

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/leverage/components/leverage-widget/components/fees.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Disclosure } from '@headlessui/react'
 import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/20/solid'
 import clsx from 'clsx'
 
-import { LeverageType } from '@/app/leverage/provider'
+import { LeverageType } from '@/app/leverage/types'
```
Comment (janndriessen) [inline src/app/leverage/components/leverage-widget/components/fees.tsx:6]: Just moved to types file

File: src/app/leverage/components/leverage-widget/index.tsx [hunk 2]
```diff
@@ -99,6 +104,7 @@ export function LeverageWidget(props: LeverageWidgetProps) {
       <BuySellSelector isMinting={isMinting} onClick={toggleIsMinting} />
       <LeverageSelector
         selectedTye={leverageType}
+        supportedTypes={supportedLeverageTypes}
```
Comment (janndriessen) [inline src/app/leverage/components/leverage-widget/index.tsx:107]: Mainnet only supports `2x`. So this makes sure the selector has the according buttons.

File: src/app/leverage/components/selectors/network-selector.tsx [hunk 3]
```diff
@@ -42,7 +42,7 @@ export function NetworkSelector(props: NetworkSelectorProps) {
           onClick={() => {
             props.onSelectNetwork(BASE.chainId)
           }}
-        />
+        /> */}
```
Comment (janndriessen) [inline src/app/leverage/components/selectors/network-selector.tsx:45]: Hiding Base until we need it. Adding Mainnet in first position according to extra ticket.

File: src/app/leverage/components/your-tokens.tsx [hunk 4]
```diff
@@ -26,17 +29,17 @@ export function YourTokens() {
...
   const handleCloseClick = (token: EnrichedToken) => {
     if (isMinting) toggleIsMinting()
 
-    const leverageBaseToken = getLeverageBaseToken(token)
+    const leverageBaseToken = getLeverageBaseToken(token.symbol)
     if (leverageBaseToken === null) return
     onSelectBaseToken(leverageBaseToken.symbol)
 
-    const leverageType = getLeverageType(token)
+    const leverageType = getLeverageType(token.symbol)
```
Comment (janndriessen) [inline src/app/leverage/components/your-tokens.tsx:42]: Made all these functions to use only the symbol as they didn't really usy anything else from `Token`

File: src/app/leverage/constants.ts [hunk 5]
```diff
@@ -22,6 +36,29 @@ export const btcLeverageTokens = [
...
+    const leverageType = getLeverageType(token.symbol)
+    if (!baseToken || !address || leverageType === null) {
+      return null
+    }
+    return {
+      ...token,
+      address,
+      baseToken: baseToken.symbol,
+      leverageType,
+    }
```
Comment (janndriessen) [inline src/app/leverage/constants.ts:52]: Add some props that are needed throughout the UI

File: src/app/leverage/utils/get-leverage-type.ts [hunk 6]
```diff
@@ -5,29 +5,22 @@ import {
...
+    case IndexCoopEthereum2xIndex.symbol:
+      return LeverageType.Long2x
+    case IndexCoopBitcoin3xIndex.symbol:
+    case IndexCoopEthereum3xIndex.symbol:
+      return LeverageType.Long3x
+    case IndexCoopInverseBitcoinIndex.symbol:
+    case IndexCoopInverseEthereumIndex.symbol:
+      return LeverageType.Short
+    default:
+      return null
```
Comment (janndriessen) [inline src/app/leverage/utils/get-leverage-type.ts:24]: Just copying here refactoring. Should work as before.

File: src/app/leverage/provider.tsx [hunk 7]
```diff
@@ -166,42 +130,32 @@ export function LeverageProvider(props: { children: any }) {
...
-        case LeverageType.Short:
-          return IndexCoopInverseBitcoinIndex
-      }
-    }
-  }, [baseToken, leverageType])
+    return indexTokens.filter(
+      (token) =>
+        token.baseToken === baseToken.symbol &&
+        token.leverageType === leverageType,
+    )[0]
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:151]: nit: could use `find` here instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: src/app/leverage/provider.tsx [hunk 8]
```diff
@@ -403,6 +398,21 @@ export function LeverageProvider(props: { children: any }) {
...
+    setInputToken(ETH)
+    setOutputToken(IndexCoopEthereum2xIndex)
+    setLeverageType(LeverageType.Long2x)
+    setQuoteResult({
+      type: QuoteType.flashmint,
+      isAvailable: true,
+      quote: null,
+      error: null,
+    })
+  }, [chainId])
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:414]: :100:

**Review by 0xonramp — COMMENTED** [review]
LGTM

File: src/app/leverage/provider.tsx (hunk 7 above)
Comment (janndriessen) [reply src/app/leverage/provider.tsx:155]: #classic I was thinking about how it was called and then forgot to check later ;D

**Pull request merged**